cmd/waku/server/rest/relay.go

```diff
@@ -112,36 +135,21 @@ func (r *RelayService) postV1Subscriptions(w http.ResponseWriter, req *http.Requ
 }
 
 func (r *RelayService) getV1Messages(w http.ResponseWriter, req *http.Request) {
-	topic := topicFromPath(w, req, "topic", r.log)
-	if topic == "" {
+	pubsubTopic := topicFromPath(w, req, "topic", r.log)
+	if pubsubTopic == "" {
 		return
 	}
+
 	//TODO: Update the API to also take a contentTopic since relay now supports filtering based on contentTopic as well.
-	sub, err := r.node.Relay().GetSubscriptionWithPubsubTopic(topic, "")
+	messages, err := r.cache.GetMessagesWithPubsubTopic(pubsubTopic)
 	if err != nil {
 		w.WriteHeader(http.StatusNotFound)
 		_, err = w.Write([]byte("not subscribed to topic"))
 		r.log.Error("writing response", zap.Error(err))
 		return
 	}
-	var response []*pb.WakuMessage
```

Comment:
Can't we just run a loop here until the channel is drained or we reached maxCacheCapacity and return?
This way we can avoid maintaining this additional cache?

Reply:
The difficulty with trying that approach is that this concept of draining a channel is not something that is used in status-go. Reading from channels is a blocking operation up to the moment you close the channel, or add some timeout like:

```
ctx, cancel := context.WithTimeout(request.ctx, 2 * time.Second)
defer cancel()
var messages []*pb.WakuMessage
for {
    select {
           case msg := <-myCh:
                         messages = append(messages, msg)
           case <-ctx.Done():
                  return messages
    }
}
```
With the drawback of having the requests potentially take more time than required when there is no messages in the channel.

Reply:
Ok, Got it.... how about something like this? 
Wouldn't this take care of the case where channel doesn't have anything to be read, we return or once we hit cache capacity we return. I may be missing something here though.

```
var messages []*pb.WakuMessage
for {
    if len(messages) > cacheCapacity {
     break
    }
    select {
           case msg := <-myCh:
                         messages = append(messages, msg)
           case <-ctx.Done():
                 break
           default:
                 break

    }
}
 return messages
```

Reply:
I'd have to test it, but I believe it would still suffer the same delay, in case the number of messages you retrieve from the channel are less than the cacheCapacity, and also because the length of the function execution would depend on the request context.

So for example if you have a cache limit of  20, and only have received 3 messages, you'd have to wait until `ctx.Done()` is triggered, and then, probably there's no one to reply, since the requestor will have closed the connection by then

Reply:
`So for example if you have a cache limit of 20, and only have received 3 messages, you'd have to wait until ctx.Done() is triggered, and then, probably there's no one to reply, since the requestor will have closed the connection by then`

Wouldn't the default case hit once channel is empty? That should take care of not having to wait till end of context? I had done this change and tested with defaultTopic and i could see 30 messages(which is the default capacity set) returned in 1 call and that too without waiting for request context to be done. 
So looks like this approach works and is simple without having to maintain additional cache layer.

BTW, we may have to increase the default cache capacity from 30 to a higher number such as 1000 or something as underlying subscription's default is that if not set. But since in REST we are setting it to 30, it is very low.